extension/test/e2e/fixtures/3ds-v2-make-payment-form.html

```diff
@@ -220,13 +220,26 @@
         console.log(component)
       }
 
+      function generateHexString() {
+        // Generate 4-digit hex string
+        const randomReferenceNumber = Math.floor(Math.random() * 9999)
+        return randomReferenceNumber.toString(16).padStart(4, '0')
+      }
+
+      function generateReference() {
+        // Generate 16-digit hex string with format xxxx-xxxx-xxxx-xxxx
+        let randomReferenceNumber = ''
+        for (let index = 1; index <= 4; index++) {
+          randomReferenceNumber = randomReferenceNumber
+            .concat(generateHexString())
+            .concat('-')
+        }
+        return randomReferenceNumber.substr(0, randomReferenceNumber.length - 1)
+      }
+
       function handleOnSubmit(state, component) {
-        console.log('handleOnSubmit')
-        console.log(state)
-        console.log(component)
         const makePaymentRequest = state.data
-        const orderReference = new Date().getMilliseconds()
-        makePaymentRequest.reference = orderReference
+        makePaymentRequest.reference = generateReference()
```

Comment:
Why not just doing `new Date().getTime()`? This will give you a unique number.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/735/commits/4dcba48499809674402d1e1ca3439810d5e959fa